rusty-merkle-tree/src/merkle_tree.rs

```diff
@@ -0,0 +1,171 @@
+extern crate crypto;
+use::crypto::sha3::Sha3;
+use crypto::digest::Digest;
+#[derive(Debug, PartialEq, Clone)]
+pub enum ArrayError{
+    Empty,
+    NotPowerOfTwo
+}
+#[derive(Debug, PartialEq, Clone)]
+pub struct MerkleTree {
+    input: Vec<String>,
+    leaves: Vec<String>
+}
+const SMALL_ARRAY: usize = 1;
+impl MerkleTree {
+    //  This function creates the MerkleTree struct
+    // verying the input array
+    fn new(leaves_array: Vec<String>) -> Result<MerkleTree, ArrayError> {
+        if let Err(error) = MerkleTree::verify_input(&leaves_array) {
+            return Err(error);
+        }
```

Comment:
You can use ? here

MerkleTree::verify_input(&leaves_array)?

Reply:
It returns Result<(), ArrayError>, how can I apply that here?

Reply:
Both approaches do the same thing, the ? is more concise